### ISSUE TITLE
Log attempted usernames and passwords

### ISSUE THREAD

**Issue by obilodeau** [GoSecure/pyrdp#293]
Right now we log successful username and password authentication with credential heuristics. I would like us to log attempted usernames and passwords as well. In both non-NLA and NLA contexts.

**Comment by xshill** [comment]
I started working on this in non-NLA situations. It works by spoofing a request for the RDPDR channel if it's not requested by the client. This way we can always rely on the RDPDR channel to detect when the user has logged on, unless it's disabled by the server.

Current progress on this is on https://github.com/GoSecure/pyrdp/tree/log-credentials-non-nla. However, I have tried the following RDP clients and all of them requested the RDPDR channel even though I wasn't sharing anything, which makes it hard to test.

- mstsc
- rdesktop
- remmina
- xfreerdp
- krdc
- xrdp

I also tried Vinagre and it doesn't request any virtual channel, which makes the spoofing implementation a bit more complicated.

I think for now the benefit of implement this for non-NLA is questionable since most clients seem to request it no matter what, so I'll focus on an NLA implementation.

Remaining work for non-NLA: send spoofed client RDPDR responses until we get the `PAKID_CORE_USER_LOGGEDON` packet and then ignore everything else.

**Comment by xshill** [comment]
I actually don't think it will be possible to log invalid credentials that were attempted with NLA. I was looking at the protocol again this morning and realised that we need to have a valid password to even go past the first 3 steps of NLA, which is the session key exchange.

In the protocol, the client and server negotiate a key that will be used to encrypt information on top of TLS (the session key). The server sends a challenge, the clients encrypts it with their password hash and sends the result. If the password is invalid, the response (and thus the session key) will be invalid and the protocol will fail at that point rather than during the credential exchange.

Indeed, I tried logging all the NLA messages with a valid and an invalid password. For the valid one I got all 5 messages needed by the protocol, whereas with an invalid password I only got 4 (3 + 1 error message), so the client never got to the final step.

![image](https://user-images.githubusercontent.com/18492633/111638055-5c837b00-87d0-11eb-9585-373001f57f66.png)
